### PR TITLE
[integration] bugfix for PKO-85: correctly attribute (Cluster)ObjectSlices to their owning (Cluster)ObjectSets

### PR DESCRIPTION
**Pull request by erdii**
This PR
- changes the signature of `testNamespacedAndCluster`'s postCheck arg to pass the namespace as a string instead of only a boolean (empty string == not namespaced)
- adds `assertAmountOfSliceObjectsControlledPerObjectSet` test helper which attributes (Cluster)ObjectSlices to their controlling (Cluster)ObjectSet via owner references
- changes `TestPackage_simpleWithSlices` and `TestPackage_simpleWithoutSlices` to use `assertAmountOfSliceObjectsControlledPerObjectSet`.

Signed-off-by: Josh Gwosdz <jgwosdz@redhat.com>

### Change Type

Bug Fix

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [x] Relevant documentation has been updated.

### CODE REVIEW

File: integration/package-operator/package_test.go [hunk 1]
```diff
@@ -228,15 +206,15 @@ func TestPackage_multi(t *testing.T) {
...
 	}
 
-	postCheck := func(ctx context.Context, t *testing.T, namespaced bool) {
+	postCheck := func(ctx context.Context, t *testing.T, namespace string) {
 		t.Helper()
 
 		var pkgBE, pkgFE client.Object
 		var ns string
-		if namespaced {
+		if len(namespace) != 0 {
```
Comment (eqrx) [inline integration/package-operator/package_test.go:214]:
```suggestion
		if len(namespace) != "" {
```

File: integration/package-operator/package_test.go [hunk 2]
```diff
@@ -264,12 +242,12 @@ func TestPackage_cel(t *testing.T) {
...
 	}
 
-	postCheck := func(ctx context.Context, t *testing.T, namespaced bool) {
+	postCheck := func(ctx context.Context, t *testing.T, namespace string) {
 		t.Helper()
 
 		var ns string
-		if namespaced {
-			ns = "default"
+		if len(namespace) != 0 {
```
Comment (eqrx) [inline integration/package-operator/package_test.go:249]:
```suggestion
		if len(namespace) != "" {
```

**Review by eqrx — CHANGES_REQUESTED** [review]

**Review by eqrx — APPROVED** [review]

**Comment by erdii** [comment]
https://github.com/package-operator/package-operator/actions/runs/8848911618/job/24299671181?pr=1118#step:5:375

> 📦 package-operator.run/integration/package-operator
> panic: PKO_TEST_SUCCESS_PACKAGE_IMAGE not set!

> goroutine 1 [running]:
> package-operator.run/integration/package-operator.init.0()
> 	/home/runner/work/package-operator/package-operator/integration/package-operator/environment.go:61 +0x72d

**Comment by erdii** [comment]
I have solved the above problem by adding `//go:build integration || integration_hypershift` to the top of the renamed file.

**Comment by erdii** [comment]
> codecov/project — 60.14% (-0.88%) compared to b8552e1

I vote for "force merge" after accepting review.

**Review by eqrx — APPROVED** [review]

**Pull request merged**